In [1]:
import sklearn
import numpy as np
import pandas as pd

from pandas import Series;

In [2]:
validation = pd.read_csv("validation.csv",skiprows = [0], header = None)
target = pd.read_csv("train.csv", skiprows = [0], header = None )

In [3]:
validation_p = validation[1].str.split(" ", 5, True)

In [4]:
vali = pd.concat( [validation[0], validation_p], axis=1,join='inner')

In [5]:
vali.columns = ["image_id", "class", "bbox","bbox0", "bbox1", "bbox2", "bbox3"]

In [6]:
target.columns = ["image_id", "class1","class", "bbox","bbox0", "bbox1", "bbox2", "bbox3"]


In [7]:
comp = pd.merge(target[["image_id", "class"]], vali[["image_id", "class"]], how = 'right', on = "image_id")

In [8]:
comp.drop_duplicates('image_id', keep='first', inplace=False)

image_id  class_x class_y
0     9f709e4b36665cf43eabf6fbe7e014ce       14      14
3     aff3ad83a6b6a86cb6ff098d920b2da9       14      14
6     ad413beaa010e38e1f06dad791e3c81c       14      14
9     81b84cb547ada6c0a05c8c6dea989b31       14      14
12    0f48cb3124f7e8c3a849f120c36dcda5       14       0
...                                ...      ...     ...
9575  7c0ddf911bdb68fca14d7172486827cb       14      14
9578  209e3a5c73c1195d805dd25d086b3c6d       14      14
9581  1bea772246421c37929f8cbc43dba729       14      14
9584  ae86eabab95525b41b8e79883ff1cef9       14      14
9587  839b2311ffad9bdd9aa98c74aad6e251       14      14

[3000 rows x 3 columns]

In [9]:
x_pred = np.array(comp['class_y'])
x_true = np.array(comp['class_x'])

In [10]:
y_pred = x_pred.astype(np.int64)


In [11]:
from sklearn.metrics import accuracy_score

accuracy_score(x_true, y_pred)

0.7763295099061522

In [12]:
from sklearn.metrics import precision_score
from sklearn.metrics import recall_score

In [13]:
print(precision_score(x_true, y_pred, average='macro')) 
print(precision_score(x_true, y_pred, average='micro')) 
print(precision_score(x_true, y_pred, average='weighted')) 

0.15805212076606334
0.7763295099061522
0.7956783860500918


/Users/chaoranhuang/Library/Python/3.8/lib/python/site-packages/sklearn/metrics/_classification.py:1245: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
/Users/chaoranhuang/Library/Python/3.8/lib/python/site-packages/sklearn/metrics/_classification.py:1245: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))


In [14]:
print(recall_score(x_true, y_pred, average='macro'))

0.15481837790408337



|  faster rcnn  | Precision(marco)  | Precision (micro)  | Precision(weighted) | recall |
|  :----  | ----  | ----  | ----  |  ----  |
|   | 0.1580 | 0.7763 | 0.7956 | 0.1548 |
